src/main/java/com/programmers/springweekly/repository/customer/JdbcTemplateCustomerRepository.java

```diff
@@ -0,0 +1,114 @@
+
+@Repository
+@Profile({"test", "local"})
+public class JdbcTemplateCustomerRepository implements CustomerRepository {
+
+    private NamedParameterJdbcTemplate template;
+
+    public JdbcTemplateCustomerRepository(DataSource dataSource) {
+        this.template = new NamedParameterJdbcTemplate(dataSource);
+    }
+
+    @Override
+    public Customer save(Customer customer) {
+        String sql = "insert into customers values (:customerId, :customerName, :customerEmail, :customerType)";
+
+        SqlParameterSource param = new MapSqlParameterSource()
+                .addValue("customerId", customer.getCustomerId().toString())
+                .addValue("customerName", customer.getCustomerName())
+                .addValue("customerEmail", customer.getCustomerEmail())
+                .addValue("customerType", customer.getCustomerType().toString());
+
+        template.update(sql, param);
+
+        return customer;
+    }
+
+    @Override
+    public void update(Customer customer) {
+        String sql = "update customers set customer_name = :customerName, customer_email = :customerEmail, customer_type = :customerType where customer_id = :customerId";
+
+        SqlParameterSource param = new MapSqlParameterSource()
+                .addValue("customerName", customer.getCustomerName())
+                .addValue("customerEmail", customer.getCustomerEmail())
+                .addValue("customerType", customer.getCustomerType().toString())
+                .addValue("customerId", customer.getCustomerId().toString());
+
+        template.update(sql, param);
+    }
+
+    @Override
+    public Optional<Customer> findById(UUID customerId) {
+        String sql = "select * from customers where customer_id = :customerId";
+
+        SqlParameterSource param = new MapSqlParameterSource()
+                .addValue("customerId", customerId.toString());
+
+        Customer customer = template.queryForObject(sql, param, customerRowMapper());
+
+        return Optional.ofNullable(customer);
+    }
+
+    @Override
+    public List<Customer> findAll() {
+        String sql = "select * from customers";
+
+        return template.query(sql, customerRowMapper());
+    }
+
+    @Override
+    public List<Customer> getBlackList() {
+        String sql = "select * from customers where customer_type = :customerType";
+
+        SqlParameterSource param = new MapSqlParameterSource()
+                .addValue("customerType", CustomerType.BLACKLIST.toString());
+
+        return template.query(sql, param, customerRowMapper());
+    }
+
+    @Override
+    public void deleteById(UUID customerId) {
+        String sql = "delete from customers where customer_id = :customerId";
+
+        SqlParameterSource param = new MapSqlParameterSource()
+                .addValue("customerId", customerId.toString());
+
+        template.update(sql, param);
+    }
+
+    @Override
+    public void deleteAll() {
+        String sql = "delete from customers where 1=:one";
```

Comment:
그냥 아래 sql로 써도 될것 같아요~

```suggestion
        String sql = "delete from customers";
```

Reply:
NamedParameterJdbcTemplate을 사용할 때는 update 쿼리를 날릴 때 꼭 param을 전달해줘야해서 위와 같이 작성했었습니다! 근데 다시 생각해보니 말씀하신 것 처럼 where문 지우고 빈 파라미터 보내줘도 가능하네요! 감사합니다 ㅎㅎ!!